README.md

```diff
@@ -9,10 +9,46 @@ clone the repository.
 
 # Usage
 In order to compile a function you have to supply it as a string to the `transpile` function.
-The returned function takes the name of the function that you want to call as the first parameter.
-Any further parameters get passed to the WebAssembly instance.
+The returned value is of the type CallWrapper.
 
 ```javascript
-const exports = Transpiler.transpile('function add(a, b) { return a + b; }');
-const result = exports('add', 12, 30);
+const wrapper = new Transpiler().transpile('function add(a, b) { return a + b; }');
+wrapper.setFunctionName('add').call(21, 21);
 ```
+
+# API
+## Transpiler creation
+* new Transpiler(): `void` <br />
+  Creates a new transpiler with an empty hook class.
+* new Transpiler(hooks: `TranspilerHooks`): `void` <br />
+  Creates a new transpiler with the specified hook instance.
+
+## Transpiler manipulation
+* Transpiler#**transpile**(content: `string`): `CallWrapper` <br />
+  Transpiles the specified content.
+
+## CallWrapper manipulation
+* CallWrapper#**setFunctionName**(functionName: `string`): `CallWrapper` <br />
+  Set the name of the initial function to be called.
+* CallWrapper#**call**(...parameters: `any[]`): `any` <br />
+  Call the initial function of the transpiled Wasm code.
+
+## TranspilerHooks manipulation
+All methods are hook methods, which are called by the `Transpiler` and the `CallWrapper`. The default implementation of all these hook methods is empty, but it can be derived from and passed along to the `Transpiler`.
```

Comment:
```suggestion
The following methods are hook methods, which get called by the `Transpiler` and the `CallWrapper`. The default implementations of them are empty and can be overwritten.
```

Reply:
I had to change this a little bit, because `TranspilerHooks` now is an interface.